backends/guest/guest_svc_read.c

```diff
@@ -261,12 +266,11 @@ static int read_path(const char *path, const int is_print_raw, const char *varia
 		if (rc == SUCCESS) {
 			if (is_print_raw || esl_data_size == DEFAULT_PK_LEN)
 				print_raw((char *)esl_data, esl_data_size);
-			else if (esl_data_size >= TIMESTAMP_LEN)
+			else if (esl_data_size >= TIMESTAMP_LEN) {
+				read_timestamp(esl_data);
 				rc = print_esl_buffer(esl_data + TIMESTAMP_LEN,
 						      esl_data_size - TIMESTAMP_LEN, variable_name);
-			else
-				prlog(PR_WARNING, "WARNING: The %s database is empty.\n",
-				      variable_name);
```

Comment:
I'm a little unclear on the change to the conditionals here. I assume a variable will be completely empty if it hadn't yet stored data or freshly reset, but otherwise it should at least contain a data size of `TIMESTAMP_LEN`. Removing the else condition would remove some error cases that should handled.

It seems like there are four conditions:
1. `esl_data_size >= TIMESTAMP_LEN` ➔ normal case, variable data at least (probably) contains a timestamp
2. `0 < esl_data_size < TIMESTAMP_LEN` ➔ malformed variable contents?
3. `esl_data_size == 0` ➔ empty variable that hasn't been written to, might be worth printing that it is empty
4. `esl_data_size == -1` ➔ an error that *should* be printed, although I don't think this can be hit if `variable_from_path()` call returns `SUCCESS`.

Reply:
Note this appears again later in the function.

Reply:
I don't think conditions 1 and 4 can ever occur?  We can remove the esl_data_size != -1 check here.